Provider/src/FirebirdSql.Data.FirebirdClient/Client/Native/FbClientFactory.cs

```diff
@@ -74,39 +75,17 @@ public static IFbClient GetFbClient(string dllName)
 			}
 
 			IFbClient fbClient;
-
-			// First, try to get the IFbClient from the cache.
-			lock (cache)
+			if (cache.TryGetValue(dllName, out fbClient))
```

Comment:
It is better to use GetOrAdd method here:
https://msdn.microsoft.com/ru-ru/library/ee378676(v=vs.110).aspx